BazelExtensions/xcode_configuration_provider.bzl

```diff
@@ -62,3 +62,38 @@ def _target_config_aspect_impl(itarget, ctx):
 target_config_aspect = aspect(
     implementation=_target_config_aspect_impl, attr_aspects=["*"]
 )
+
+
+XcodeBuildSourceInfo = provider(
+    fields={
+        "values": """The values of source files
+"""
+    }
+)
+
+def _extract_generated_sources(target):
+    """ Collects all of the generated source files"""
+    if not hasattr(target, "objc"):
+        return []
+    objc_provider = target.objc
+    if hasattr(objc_provider, "source") and hasattr(objc_provider, "header"):
```

Comment:
does it need to have both `attrs` or should we add to `all_files` if we contain a `header`?

Reply:
Currently, this is referring to the `objc` provider. Currently it provides the `header` and `source` `attr` to the aspect and there isn't one without the other